components/director/internal/domain/webhook/service.go

```diff
@@ -188,6 +202,75 @@ func (s *service) Delete(ctx context.Context, id string, objectType model.Webhoo
 	return s.webhookRepo.Delete(ctx, webhook.ID)
 }
 
+// EnrichWebhooksWithTenantMappingWebhooks enriches webhook inputs with tenant mapping webhooks based on the tenant mapping
+// configuration. In order to be enriched, the input webhooks should contain Version, URL and Mode
+func (s *service) EnrichWebhooksWithTenantMappingWebhooks(in []*graphql.WebhookInput) ([]*graphql.WebhookInput, error) {
+	webhooks := make([]*graphql.WebhookInput, 0)
+	for _, w := range in {
+		if w.Version == nil {
+			webhooks = append(webhooks, w)
+			continue
+		}
+
+		if w.URL == nil || w.Mode == nil {
+			return nil, errors.New("url and mode are required fields when version is provided")
+		}
+		tenantMappingWebhooks, err := s.getTenantMappingWebhooks(w.Mode.String(), *w.Version)
+		if err != nil {
+			return nil, err
+		}
+		for _, tenantMappingWebhook := range tenantMappingWebhooks {
+			urlTemplate := *tenantMappingWebhook.URLTemplate
+			if strings.Contains(urlTemplate, "%s") {
+				urlTemplate = fmt.Sprintf(*tenantMappingWebhook.URLTemplate, *w.URL)
+			}
+
+			headerTemplate := *tenantMappingWebhook.HeaderTemplate
+			if *w.Mode == graphql.WebhookModeAsyncCallback && strings.Contains(headerTemplate, "%s") {
+				headerTemplate = fmt.Sprintf(*tenantMappingWebhook.HeaderTemplate, s.tenantMappingCallbackURL)
+			}
+			wh := &graphql.WebhookInput{
+				Type:           tenantMappingWebhook.Type,
+				Auth:           w.Auth,
+				Mode:           w.Mode,
+				URLTemplate:    &urlTemplate,
+				InputTemplate:  tenantMappingWebhook.InputTemplate,
+				HeaderTemplate: &headerTemplate,
+				OutputTemplate: tenantMappingWebhook.OutputTemplate,
+			}
+			webhooks = append(webhooks, wh)
+		}
+	}
+	return webhooks, nil
+}
+
+func (s *service) getTenantMappingWebhooks(mode, version string) ([]graphql.WebhookInput, error) {
```

Comment:
add godoc

Reply:
We do not place godoc to unexported functions